export/http.go

```diff
@@ -0,0 +1,87 @@
+package export
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/resource"
+	"google.golang.org/protobuf/proto"
+)
+
+// HTTP is an exporter that exports metrics to an
+// HTTP endpoint
+type HTTP struct {
+	address   string
+	aPIkey    string
+	aPISecret string
+	headerMap map[string]string
+	client    *http.Client
+	config    Config
+}
+
+// NewHTTP returns a new exporter agent with an HTTP exporter attached
+func NewHTTP(address string, apikey string, apisecret string, headers map[string]string, config Config) *ExporterAgent {
+	headers["Content-Type"] = "application/x-protobuf"
+
+	exporter := HTTP{
+		address:   address,
+		aPIkey:    apikey,
+		aPISecret: apisecret,
+		headerMap: headers,
+		client:    &http.Client{},
+		config:    config,
+	}
+
+	agent := newExporterAgent(exporter)
+	if err := agent.Start(exporter.config.ReportingPeriodms); err != nil {
+		panic(err)
+	}
+
+	return agent
+
+}
+
+// ExportMetrics converts the metrics to a metrics service request protobuf and
+// makes a POST request with that payload to an HTTP endpoint.
+func (e HTTP) ExportMetrics(ctx context.Context, data []*metricdata.Metric) error {
+	includeData := []*metricdata.Metric{}
+
+	for _, d := range data {
+		if matched, _ := regexp.Match(e.config.IncludeFilter, []byte(d.Descriptor.Name)); matched {
+			d.Resource, _ = resource.FromEnv(ctx)
+			includeData = append(includeData, d)
+		}
+	}
+
+	metricsRequestpb := metricsToServiceRequest(includeData)
```

Comment:
nit: keep the camelCase convention consistent, e.g. call this maybe `metricsRequestProto`